DuckDuckGoTests/MockSecureVault.swift

```diff
@@ -121,3 +141,244 @@ final class MockSecureVault: SecureVault {
     }
 
 }
+
+// MARK: - Mock Providers
+
+class MockDatabaseProvider: AutofillDatabaseProvider {
```

Comment:
I'd like to move these mocks into a support package in BSK, so that they can be reused by iOS and macOS. Will look at that on Monday.